### PR TITLE
Add nanoserver Dockerfiles for 1709 release

### LINKED ISSUE #298: Create Nano Server images for the 1709 release
Nano Server based images should be created and available when [Windows Server 1709](https://blogs.technet.microsoft.com/windowsserver/2017/09/25/faster-releases-for-a-faster-world-first-semi-annual-release-of-windows-server/) releases.  The existing Nano Server images must be maintained as 1709 images are not backwards compatible.

### PR DESCRIPTION
**Pull request by MichaelSimons**
Fixes #298 

Remaining changes:

- [x] Enable cleanup steps on Windows-1709 build leg (can't enable until base images are GA)
- [x] Replace `msimons` Docker registry references with `microsoft`

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
     ],
     "windows": [
-      "powershell -NoProfile -Command .\\test\\run-test.ps1 -Filter $(Filter)"
+      "powershell -NoProfile -Command .\\test\\run-test.ps1 -Filter $(Filter) -OS $(OS)"
     ]
   },
   "repos": [
     {
-      "name": "microsoft/dotnet",
+      "name": "msimons/dotnet",
```
Comment (richlander) [inline manifest.json]: What's this change for?

File: test/Microsoft.DotNet.Docker.Tests/ImageTests.cs [hunk 2]
```diff
@@ -204,7 +227,7 @@ private void VerifyRuntimeDepsImage_SelfContainedApp(ImageDescriptor imageDescri
             string imageVersion, DotNetImageType imageType, string osVariant, bool isArm = false)
         {
             string variantName = Enum.GetName(typeof(DotNetImageType), imageType).ToLowerInvariant().Replace('_', '-');
-            string imageName = $"microsoft/dotnet:{imageVersion}-{variantName}";
+            string imageName = $"msimons/dotnet:{imageVersion}-{variantName}";
```
Comment (richlander) [inline test/Microsoft.DotNet.Docker.Tests/ImageTests.cs]: same question here.

**Comment by richlander** [comment]
LGTM

File: manifest.json (hunk 1 above)
Comment (MichaelSimons) [reply manifest.json]: These are changes I needed to make in order to test the build infra.  I called them out in the PR comment as something that will get reverted prior to final review.

**Comment by MichaelSimons** [comment]
These changes are ready for final review.  Test images are available at msimons/dotnet.

**Pull request merged**